src/vue/modules/issuances-explorer/index.spec.js

```diff
@@ -0,0 +1,156 @@
+
+import { Wallet } from '@tokend/js-sdk'
+
+import * as Api from './_api'
+import { ErrorHandler } from '@/js/helpers/error-handler'
+
+const localVue = createLocalVue()
+localVue.use(Vuex)
+
+describe('Issuances explorer module', () => {
+  const props = {
+    config: {
+      horizonUrl: 'https://test.api.com',
+    },
+    wallet: new Wallet(
+      'test@mail.com',
+      'SCPIPHBIMPBMGN65SDGCLMRN6XYGEV7WD44AIDO7HGEYJUNDKNKEGVYE',
+      'GDIU5OQPAFPNBP75FQKMJTWSUKHTQTBTHXZWIZQR4DG4QRVJFPML6TTJ',
+      '4aadcd4eb44bb845d828c45dbd68d5d1196c3a182b08cd22f05c56fcf15b153c'
+    ),
+    shouldUpdate: false,
+  }
+
+  let store
+  let wrapper
+
+  beforeEach(() => {
+    store = new Vuex.Store({
+      modules: { 'issuances-explorer': issuancesExplorerModule },
+    })
+
+    wrapper = shallowMount(IssuancesExplorerModule, {
+      store,
+      localVue,
+      propsData: props,
+    })
+  })
+
+  describe('created hook', () => {
+    it('initializes the API', () => {
+      const spy = sinon.spy(Api, 'initApi')
+
+      shallowMount(IssuancesExplorerModule, {
+        localVue,
+        store,
+        propsData: props,
+      })
+
+      expect(spy.withArgs(props.wallet, props.config).calledOnce).to.be.true
+
+      spy.restore()
+    })
+
+    it('sets the account ID', () => {
+      const spy = sinon.spy(IssuancesExplorerModule.methods, 'setAccountId')
+
+      shallowMount(IssuancesExplorerModule, {
+        localVue,
+        store,
+        propsData: props,
+      })
+
+      expect(spy.withArgs(props.wallet.accountId).calledOnce).to.be.true
+
+      spy.restore()
+    })
+
+    it('initializes first issuances page loader', () => {
+      const spy = sinon.spy(IssuancesExplorerModule.methods, 'initFirstPageLoader')
+
+      shallowMount(IssuancesExplorerModule, {
+        localVue,
+        store,
+        propsData: props,
+      })
+
+      expect(spy.calledOnce).to.be.true
+
+      spy.restore()
+    })
+  })
+
+  describe('watcher', () => {
+    describe('shouldUpdate', () => {
+      it('should initialize page loader if the passed value is true', () => {
+        const spy = sinon.stub(wrapper.vm, 'initFirstPageLoader').resolves()
+
+        wrapper.setProps({ shouldUpdate: true })
+
+        wrapper.vm.$nextTick(_ => {
+          expect(spy.calledOnce).to.be.true
+          spy.restore()
+        })
+      })
+
+      it('should not initialize page loader if the passed value is false', () => {
+        const spy = sinon.stub(wrapper.vm, 'initFirstPageLoader').resolves()
+
+        wrapper.setProps({ shouldUpdate: false })
+
+        wrapper.vm.$nextTick(_ => {
+          expect(spy.notCalled).to.be.true
+          spy.restore()
+        })
+      })
+    })
+  })
+
+  describe('method', () => {
+    describe('loadIssuancesFirstPage', () => {
+      it('loads the account issuances', async () => {
+        const spy = sinon.stub(wrapper.vm, 'loadIssuances').resolves({
+          data: [],
```

Comment:
same about mocking the response that you don't actually need to mock.

Reply:
Updated